<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title></title>

<script src="html_support/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="html_support/bootstrap-2.3.2/css/spacelab.min.css" rel="stylesheet" />
<link href="html_support/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="html_support/bootstrap-2.3.2/js/bootstrap.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="html_support/highlight/textmate.css"
      type="text/css" />
<script src="html_support/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<link rel="stylesheet" href="html_support/custom.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">



<div id="TOC">
<ul>
<li><a href="#figure-2d-and-2e-chip-seq-vs-chip-nexus-motif-specificity">Figure 2D and 2E: ChIP-seq vs ChIP-nexus motif specificity</a><ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#dorsal">Dorsal</a></li>
<li><a href="#twist">Twist</a></li>
<li><a href="#session-information">Session information</a></li>
</ul></li>
</ul>
</div>

<div id="figure-2d-and-2e-chip-seq-vs-chip-nexus-motif-specificity" class="section level1">
<h1>Figure 2D and 2E: ChIP-seq vs ChIP-nexus motif specificity</h1>
<p><strong>Project:</strong> ChIP-nexus methods paper</p>
<p><strong>Author:</strong> <a href="mailto:jjj@stowers.org">Jeff Johnston</a></p>
<p><strong>Generated:</strong> Thu Dec 04 2014, 03:17 PM</p>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p><strong>Using top 200 peaks</strong></p>
</div>
<div id="dorsal" class="section level2">
<h2>Dorsal</h2>
<p><img src="figure_2_d_e_output/dl_profiles-1.png" title="" alt="" width="864" /></p>
<p><img src="figure_2_d_e_output/dl_nearest_motif-1.png" title="" alt="" width="864" /></p>
<!-- html table generated in R 3.1.1 by xtable 1.7-4 package -->
<!-- Thu Dec  4 15:18:14 2014 -->
<table border="1">
<tr>
<th>
variable
</th>
<th>
test
</th>
<th>
pvalue
</th>
</tr>
<tr>
<td>
within_10
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
1.819587649606e-11
</td>
</tr>
<tr>
<td>
within_20
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
3.24438312032412e-07
</td>
</tr>
<tr>
<td>
within_50
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
0.000116124473545087
</td>
</tr>
<tr>
<td>
within_100
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
0.000610563567508063
</td>
</tr>
</table>
</div>
<div id="twist" class="section level2">
<h2>Twist</h2>
<p><img src="figure_2_d_e_output/twi_profiles-1.png" title="" alt="" width="864" /></p>
<p><img src="figure_2_d_e_output/twi_nearest_motif-1.png" title="" alt="" width="864" /></p>
<!-- html table generated in R 3.1.1 by xtable 1.7-4 package -->
<!-- Thu Dec  4 15:18:20 2014 -->
<table border="1">
<tr>
<th>
variable
</th>
<th>
test
</th>
<th>
pvalue
</th>
</tr>
<tr>
<td>
within_10
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
3.02640597380651e-23
</td>
</tr>
<tr>
<td>
within_20
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
5.04322830976282e-22
</td>
</tr>
<tr>
<td>
within_50
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
1.73041130740358e-12
</td>
</tr>
<tr>
<td>
within_100
</td>
<td>
ChIP-nexus different than ChIP-seq
</td>
<td>
3.39392177873909e-11
</td>
</tr>
</table>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<p>For reproducibility, this analysis was performed with the following R/Bioconductor session:</p>
<pre><code>R version 3.1.1 (2014-07-10)
Platform: x86_64-pc-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] lattice_0.20-29                       digest_0.6.4                         
 [3] yaml_2.1.13                           dplyr_0.3.0.2                        
 [5] colorspace_1.2-4                      reshape2_1.4                         
 [7] ggplot2_1.0.0                         BSgenome.Dmelanogaster.UCSC.dm3_1.4.0
 [9] BSgenome_1.34.0                       rtracklayer_1.26.1                   
[11] Biostrings_2.34.0                     XVector_0.6.0                        
[13] GenomicRanges_1.18.1                  GenomeInfoDb_1.2.0                   
[15] IRanges_2.0.0                         S4Vectors_0.4.0                      
[17] BiocGenerics_0.12.0                   xtable_1.7-4                         

loaded via a namespace (and not attached):
 [1] assertthat_0.1          base64enc_0.1-2         BatchJobs_1.4          
 [4] BBmisc_1.7              BiocParallel_1.0.0      bitops_1.0-6           
 [7] brew_1.0-6              checkmate_1.5.0         codetools_0.2-9        
[10] DBI_0.3.1               evaluate_0.5.5          fail_1.2               
[13] foreach_1.4.2           formatR_1.0             GenomicAlignments_1.2.0
[16] gtable_0.1.2            htmltools_0.2.6         iterators_1.0.7        
[19] knitr_1.7               labeling_0.3            lazyeval_0.1.9         
[22] magrittr_1.0.1          MASS_7.3-35             munsell_0.4.2          
[25] plyr_1.8.1              proto_0.3-10            Rcpp_0.11.3            
[28] RCurl_1.95-4.3          rmarkdown_0.3.3         Rsamtools_1.18.0       
[31] RSQLite_0.11.4          scales_0.2.4            sendmailR_1.2-1        
[34] stringr_0.6.2           tools_3.1.1             XML_3.98-1.1           
[37] zlibbioc_1.12.0        </code></pre>
</div>
</div>


</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
